### PR TITLE
Suspended users are getting email notification for group messages - Fixed

### LINKED ISSUE #2727: Suspended users are getting email notification for group messages
**Describe the bug**
If a member is suspended from dashboard users page then the member is not showing in groups All member page but the suspended member getting email notification for group messages.
Please see this video: https://www.loom.com/share/779a68a6837e45fb980af10611a630b5

**To Reproduce**
Please see the above video to replicate this issue.

**Expected behavior**
Suspended member should not receive any email notification

**Screenshots**

**Environment**
My dev site and client site

**Support ticket links**
https://secure.helpscout.net/conversation/1578858124/154849?folderId=4312419

### PR DESCRIPTION
**Pull request by mahedihasannoman**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2727 .

### How to test the changes in this Pull Request:

1. Suspend a group member
2. Then send a group message to all group member
3. See the suspended member is not receiving email notification

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by chetansatasiya — APPROVED** [review]
LGTM 🚀

**Comment by mohit-buddyboss** [comment]
@mahedihasannoman 

Tested on dev branch of the platform as well as also verified on 'fix/2727'.

Issue: In this branch when user tries to send private messages to suspended members then user is able to send. Although notification do not gets generate but suspended label is not appearing, Notification message is not appearing and user have access of medium editor under suspended user thread.

**Comment by mahedihasannoman** [comment]
@mohit-buddyboss Working on this issue.

File: src/bp-messages/classes/class-bp-messages-thread.php [hunk 1]
```diff
@@ -1655,16 +1655,6 @@ public static function get( $args ) {
...
-		/**
-		 * Filters the Where SQL statement.
-		 *
-		 * @since BuddyBoss 1.5.4
-		 *
-		 * @param array $r                Array of parsed arguments for the get method.
-		 * @param array $where_conditions Where conditions SQL statement.
-		 */
-		$where_conditions = apply_filters( 'bp_recipients_recipient_get_where_conditions', $where_conditions, $r );
-
```
Comment (KartikSuthar) [inline src/bp-messages/classes/class-bp-messages-thread.php:1667]: why removed this filter its introduce from 1.5.4

**Review by KartikSuthar — CHANGES_REQUESTED** [review]
@janakkaneriya007 Can you take over this PR.

File: src/bp-messages/classes/class-bp-messages-thread.php (hunk 1 above)
Comment (mahedihasannoman) [reply src/bp-messages/classes/class-bp-messages-thread.php]: My apologies for this. I have re-added this apply_filter now. @KartikSuthar

**Review by KartikSuthar — APPROVED** [review]

**Comment by mohit-buddyboss** [comment]
@mahedihasannoman 

Tested using dev branch of the platform.
Now suspended users are not getting email notification and on getting unsuspend they are getting email notification.
Issue here : https://github.com/buddyboss/buddyboss-platform/pull/2736#issuecomment-896316788 -FIXED

Video: https://loom.com/share/26f4d5388f1c47ce92bd4e05213b2e89

Related issue is resolve now and PR is ready for release.

cc @sourabhmatolia @KartikSuthar @chetansatasiya

**Comment by JetalDobariya** [comment]
Hello,

I tested this PR in QA Approved branch , Now Suspended users are not received mail notification.

Video : https://www.loom.com/share/13e4c3934fe540e180e6b821500b608e

Thanks!
Jetal!

**Pull request merged**